src/main/java/core/basesyntax/GenericDao.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface GenericDao<T> {
```

Comment:
although overall it is a good idea to create GenericDao, for this task you will have trouble with field injection

let's create two separate interfaces